src/lib/gates.ts

```diff
@@ -149,6 +151,41 @@ function generic(
   );
 }
 
+/**
+ * **[lookup constraint](https://o1-labs.github.io/proof-systems/specs/kimchi.html?highlight=lookup%20gate#lookup)**
+ *
+ * Lookups allow you to check if a single value, or a series of values, are part of a table. The first case is useful to check for checking if a value belongs to a range (from 0 to 1,000, for example), whereas the second case is useful to check truth tables (for example, checking that three values can be found in the rows of an XOR table) or write and read from a memory vector (where one column is an index, and the other is the value stored at that index).
+ *
+ * @param tableId the [id](https://github.com/o1-labs/proof-systems/blob/master/kimchi/src/circuits/lookup/tables/mod.rs) of the lookup table.
+ * @param index0 the index of the first value to lookup.
+ * @param value0 the first value to lookup.
+ * @param index1 the index of the second value to lookup.
+ * @param value1 the second value to lookup.
+ * @param index2 the index of the third value to lookup.
+ * @param value2 the third value to lookup.
+ *
+ */
+function lookup(
```

Comment:
Maybe don't expose this to the very end user? It is a bit low level.

Reply:
gates are not currently exported from o1js! but I think they will eventually, having both low- and high-level APIs makes sense IMO